src/App.tsx

```
@@ -1,10 +1,51 @@
 /* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import {
+  todosService,
+  TodosServiceErrors,
+  todosServiceErrorText,
+  USER_ID,
+} from './api/todos';
+import cn from 'classnames';
+import { TodoItem } from './components/TodoItem';
+import { TODO_STATUS_FILTER_OPTIONS, Status } from './types/TodoStatusFilter';
+import { getFilteredTodos, Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState<string | null>(null);
+  const [selectedStatus, setSelectedStatus] = useState(Status.ALL);
+
+  const handleRemoveError = () => setError('');
+
+  useEffect(() => {
+    if (!error) {
+      return;
+    }
+
+    const timer = setTimeout(handleRemoveError, 3000);
+
+    return () => {
+      clearTimeout(timer);
+    };
+  }, [error]);
+
+  useEffect(() => {
+    todosService
+      .getTodos()
+      .then(setTodos)
+      .catch(() => {
+        setError(
+          todosServiceErrorText[TodosServiceErrors.UNABLE_TO_LOAD_TODOS],
+        );
+      })
+      .finally(() => {});
```

This empty `finally` block is unnecessary and can be removed.

---

Makes sence - finally does nothing and catch swallows the error